src/pipe_log.rs

```diff
@@ -84,92 +124,76 @@ impl PipeLog {
                 continue;
             }
 
             let file_name = file_path.file_name().unwrap().to_str().unwrap();
-            if file_name.ends_with(LOG_SUFFIX) && file_name.len() == FILE_NAME_LEN {
+            if is_log_file(file_name) {
                 let file_num = match extract_file_num(file_name) {
                     Ok(num) => num,
-                    Err(_) => {
-                        continue;
-                    }
+                    Err(_) => continue,
                 };
                 min_file_num = cmp::min(min_file_num, file_num);
                 max_file_num = cmp::max(max_file_num, file_num);
                 log_files.push(file_name.to_string());
             }
         }
 
-        // Initialize.
-        let mut pipe_log = PipeLog::new(cfg, cache_submitor);
         if log_files.is_empty() {
-            {
-                let mut manager = pipe_log.log_manager.wl();
-                let new_fd = new_log_file(&pipe_log.dir, manager.active_file_num)?;
-                manager.active_log_fd = new_fd;
-                manager.all_files.push_back(new_fd);
-            }
+            // New created pipe log, open the first log file.
+            pipe_log.log_manager.wl().new_log_file(&pipe_log.dir)?;
             pipe_log.write_header()?;
             return Ok(pipe_log);
         }
 
         log_files.sort();
-        log_files.dedup();
         if log_files.len() as u64 != max_file_num - min_file_num + 1 {
-            return Err(box_err!("Corruption occurs"));
+            return Err(box_err!("Corruption occurs on log files"));
         }
 
-        {
-            let mut manager = pipe_log.log_manager.wl();
-            manager.first_file_num = min_file_num;
-            manager.active_file_num = max_file_num;
-        }
-        pipe_log.open_all_files()?;
+        pipe_log.open_all_files(log_files, min_file_num, max_file_num)?;
         Ok(pipe_log)
     }
 
-    fn open_all_files(&mut self) -> Result<()> {
+    fn open_all_files(
+        &mut self,
+        logs: Vec<String>,
+        min_file_num: u64,
+        max_file_num: u64,
+    ) -> Result<()> {
         let mut manager = self.log_manager.wl();
-        let mut current_file = manager.first_file_num;
-        while current_file <= manager.active_file_num {
-            let mut path = PathBuf::from(&self.dir);
-            path.push(generate_file_name(current_file));
+        manager.first_file_num = min_file_num;
+        manager.active_file_num = max_file_num;
 
-            let oflag = if current_file < manager.active_file_num {
-                // Open inactive files with readonly mode.
-                OFlag::O_RDONLY
-            } else {
-                // Open active file with readwrite mode.
-                OFlag::O_RDWR
-            };
-            let fd = fcntl::open(&path, oflag, Mode::S_IRWXU)
-                .map_err(|e| raft::StorageError::Other(e.into()))?;
+        for file_name in &logs[0..logs.len() - 1] {
```

Comment:
I think `&logs` is enough?

Reply:
the last one file is handled after the for loop. That file needs to be opened with flag `RDWR`.